### PR TITLE
BERT training on IMDB movie sentiment

### LINKED ISSUE #201: [BUG] Convert kernel cannot correctly cast type.
**🐛 Bug**

Current `convert` kernel rely on implicit conversions to cast tensor type. But it might lead unexpected result if we have different output type with the same input type in a single project, eg:
```cpp
__device__ __forceinline__ int64_t convert(float x0)
{
    return x0;
}
__device__ __forceinline__ float convert(float x0)
{
    return x0;
}
```

**Additional context**

we should leverage cuda casting func in this case: https://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH__INTRINSIC__CAST.html

### PR DESCRIPTION
**Pull request by mzmssg**
This PR is to train huggingface bert on a real dataset(IMDB movie reviews), including:
1. Scripts to train BERT in NNFusion on IMDB data
2. Add dedicate backward op/kernel for `Power`
3. Fix static linking issue
4. Backward translator for `Concat`
5. ONNX translator for `NonZero`
6. Defaulting sgd lr to 1e-4
7. AD support for multi-output graph
8. Disabled weight NAN checker to speed up execution
9. Trainer support customized codegen flags
10. Add code snippet of PyTorch BERT training
11. Fix #201
12. Handle optimizer deps, caused by #200 
13. Ignore control flow in blockfusion
14. revert #198 

## Limitation
1. Current training used a fixed lr sgd optimizer(no support for customizing optimizers), and blockfusion fusion disabled(cause weight NAN), dropout disabled(kernel not ready)

## Usage
See `src/python/example/bert.py:train_bert()`

## Test
**In NNF root folder**, `python3 src/python/example/bert.py`, which train a classifcation model(3 epoch, about 1~2 hour), finally outputs prediction as:
image

### CODE REVIEW

**Review by wenxcs — APPROVED** [review]

**Comment by wenxcs** [comment]
@mzmssg  Do we have a click-to-run script for imdb ds test?

**Comment by mzmssg** [comment]
@wenxcs 
Yes, `src/python/example/bert.py` could be run for e2e training & evaluation

**Review by wenxcs — APPROVED** [review]
